contrib/scripts/createmodule.py

```diff
@@ -36,10 +36,21 @@
     parser.print_usage()
     exit(1)
 
+# Determine if running environment is based on cmd.exe or not
+def iscmdshell():
+    import platform
+    return True if platform.system().lower() == 'windows' else False
+    
 # Return environment after a command
 def getenv(cmd = ':'):
     env = {}
-    p = Popen(cmd + ";env", shell=True, stdout=PIPE, stderr=PIPE, universal_newlines=True)
+    if iscmdshell():
+        # ':' command not supported by cmd.exe
+        if cmd == ':':
+            cmd = '@echo off'
+        p = Popen(cmd + " & set", shell=True, stdout=PIPE, stderr=PIPE, universal_newlines=True)
+    else:
+        p = Popen(cmd + ";env", shell=True, stdout=PIPE, stderr=PIPE, universal_newlines=True)
```

Comment:
As `Popen` call has the same option in both branches, it seems preferable just to adapt `cmd` in condition then having a single `Popen` call.